tools/shell/shell_toolchain.bzl

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Configure the shell toolchain on the local machine."""
+
+def _shell_toolchain_impl(ctx):
+  """shell_toolchain rule implementation."""
+  return [platform_common.ToolchainInfo(path = ctx.attr.path)]
+
+def _is_windows(repository_ctx):
```

Comment:
Most of this file is the repository rule that autodetects the shell toolchain. Is it worth splitting the definition of the actual shell_toolchain rule to a separate .bzl file, so that people defining standalone shell toolchains aren't loading all of this?

Reply:
Good point, done.